lib/hyrax/engine.rb

```diff
@@ -32,10 +32,22 @@ class Engine < ::Rails::Engine
     end
 
     config.after_initialize do
-      begin
+      can_connect = begin
+        ActiveRecord::Base.connection
+        true
+      rescue StandardError
+        false
+      end
+
+      can_persist = can_connect && begin
         Hyrax.config.persist_registered_roles!
         Rails.logger.info("Hyrax::Engine.after_initialize - persisting registered roles!")
+        true
       rescue ActiveRecord::StatementInvalid
```

Comment:
What makes this invalid?

Reply:
Nothing makes this invalid. The old behavior was to call `Hyrax.config.persist_registered_roles!` and catch `ActiveRecord::StatementInvalid`. I'm just using `true` and `false` as an indicator of whether persistence went off without a hitch (under the pre-PR rubric).

The warning used to be logged in the rescue clause itself. Now I'm using a two-stage success/failure boolean to determine if the warning should be logged.

Reply:
Sorry, I misread the question (and the line it was directed at). I believe this catch is what allows `db:migrate` to happen. If the table doesn't exist yet, this would raise `ActiveRecord::StatementInvalid` on load. It's part of the preexisting code, though.

Reply:
Okay, that makes sense.